enarx-keep-sev/src/vm/mem.rs

```diff
@@ -0,0 +1,32 @@
+// SPDX-License-Identifier: Apache-2.0
+
+pub use kvm_bindings::kvm_userspace_memory_region as KvmUserMemoryRegion;
+use mmap::Unmap;
+
+pub struct Region {
+    _kvm_region: KvmUserMemoryRegion,
+    _unmap: Unmap,
+}
+
+impl Region {
+    pub fn new(kvm_region: KvmUserMemoryRegion, unmap: Unmap) -> Self {
+        Self {
+            _kvm_region: kvm_region,
+            _unmap: unmap,
+        }
+    }
+}
+
+pub struct AddressSpace {
+    regions: Vec<Region>,
+}
+
+impl AddressSpace {
+    pub fn new() -> Self {
+        Self { regions: vec![] }
+    }
+
+    pub fn add_region(&mut self, region: Region) {
+        self.regions.push(region);
+    }
```

Comment:
I think the general strategy of Rust (maybe I'm wrong) is to just make `regions` public rather than focusing on encapsulation.

Reply:
Which basically means that: `type AddressSpace = Vec<Region>`.